parser/parser_test.go

```diff
@@ -192,3 +192,86 @@ sub vcl_recv {
 	}
 	assert(t, vcl, expect)
 }
+
+func TestAllCommentPositions(t *testing.T) {
+	input := `
+// subroutine leading
+sub /* subroutine ident leading */ vcl_recv {
```

Comment:
For completeness it would be good to include a subroutine block leading comment.

```
sub /* subroutine ident leading */ vcl_recv /* subroutine block leading */ {
```

Also maybe before / after the parens of the if statement condition.

Reply:
Okay, it's easy to handle it - done.